src/Microsoft.AspNetCore.Sockets/Transports/ServerSentEventsTransport.cs

```diff
@@ -27,43 +29,43 @@ public ServerSentEventsTransport(ReadableChannel<Message> application, ILoggerFa
             context.Response.ContentType = "text/event-stream";
             context.Response.Headers["Cache-Control"] = "no-cache";
             context.Response.Headers["Content-Encoding"] = "identity";
-            await context.Response.Body.FlushAsync();
+
+            var pipe = context.Response.Body.AsPipelineWriter();
+
+            await pipe.Alloc().FlushAsync();
 
             try
             {
                 while (await _application.WaitToReadAsync(token))
                 {
+                    var buffer = pipe.Alloc();
                     while (_application.TryRead(out var message))
                     {
-                        await Send(context, message);
+                        if (!ServerSentEventsMessageFormatter.TryFormatMessage(message, buffer.Memory.Span, out var written))
+                        {
+                            // We need to expand the buffer
+                            // REVIEW: I'm not sure I fully understand the "right" pattern here...
+                            buffer.Ensure(LongPollingTransport.MaxBufferSize);
```

Comment:
Yeah, I agree. I need to work with @davidfowl to figure out the best way to handle this. The APIs for formatting messages don't return any kind of hint as to how much more data is needed.